### PR TITLE
Jhancock/use elisa microservice

### PR DESCRIPTION
**Pull request by JonathanHancock0**
A microservice will now handle ELisA authentification.

### CODE REVIEW

File: src/nanorc/logbook.py [hunk 1]
```diff
@@ -66,55 +64,26 @@ def _start_new_message_thread(self):
...
-                self.log.error(f"ELisA logbook: {str(ex)}")
-                self.log.error(answer)
-                raise ex
-
-            except Exception as e:
-                self.log.error(f'Exception thrown while inserting data in elisa:')
-                self.log.error(e)
-                import logging
+        #Since NAnoRC is sending these messages, is the system always DAQ?
+        data = {'author':user, 'title':subject, 'body':body, 'command':command, 'systems':["DAQ"]}
```
Comment (TiagoTAlves) [inline src/nanorc/logbook.py]: daq Not DAQ (I think this is what worked on drunc)

File: src/nanorc/logbook.py [hunk 2]
```diff
@@ -66,55 +64,26 @@ def _start_new_message_thread(self):
...
+
+        if not self.current_id:
+            r = requests.post(f'{self.socket}/v1/elisaLogbook/new_message', auth=(fooUsr,barPass), data=data)
```
Comment (TiagoTAlves) [inline src/nanorc/logbook.py]: Is there a way to not have the authorisation pull from nanorc.json instead?

File: src/nanorc/logbook.py [hunk 3]
```diff
@@ -66,55 +64,26 @@ def _start_new_message_thread(self):
...
+            data["id"=self.current_id]
+            r = requests.put(f'{self.socket}/v1/elisaLogbook/reply_to_message', auth=(fooUsr,barPass), data=data)
+
+        response = r.json()
+        if r.status_code != 201:
+            self.log.error(f'Exception thrown while inserting data in elisa:')
+            e = (response['response'])
+            self.log.error(e)
+            import logging
                 if logging.DEBUG >= logging.root.level:
```
Comment (TiagoTAlves) [inline src/nanorc/logbook.py:84]: Indentation error

File: src/nanorc/logbook.py [hunk 4]
```diff
@@ -66,55 +64,26 @@ def _start_new_message_thread(self):
...
-                self.log.error(f'Exception thrown while inserting data in elisa:')
-                self.log.error(e)
-                import logging
+        #Since NAnoRC is sending these messages, is the system always DAQ?
+        data = {'author':user, 'title':subject, 'body':body, 'command':command, 'systems':["DAQ"]}
+
+        if not self.current_id:
+            r = requests.post(f'{self.socket}/v1/elisaLogbook/new_message', auth=(fooUsr,barPass), data=data)
+        else:
+            data["id"=self.current_id]
```
Comment (TiagoTAlves) [inline src/nanorc/logbook.py]: Is it meant to be "data =" ?

**Review by TiagoTAlves — CHANGES_REQUESTED** [review]
Core.py line 16 ElisaLogbook to ElisaHandler because you changed it in logbook.py

File: src/nanorc/logbook.py (hunk 2 above)
Comment (TiagoTAlves) [reply src/nanorc/logbook.py]: Also you forgot the "/" slash at the end "/v1/elisaLogbook/new_message"

File: src/nanorc/logbook.py (hunk 1 above)
Comment (JonathanHancock0) [reply src/nanorc/logbook.py]: Sounds right, changed it

File: src/nanorc/logbook.py (hunk 3 above)
Comment (JonathanHancock0) [reply src/nanorc/logbook.py:84]: I guess mentally I was treating the import like an if, fixed

File: src/nanorc/logbook.py (hunk 4 above)
Comment (JonathanHancock0) [reply src/nanorc/logbook.py]: It is indeed

File: src/nanorc/logbook.py (hunk 2 above)
Comment (JonathanHancock0) [reply src/nanorc/logbook.py]: I've changed the __main_np04_.py to get the (user,pass) tuple from .nanorc, similarly to how it gets the socket. (I also added the /)

**Review by TiagoTAlves — APPROVED** [review]

**Comment by TiagoTAlves** [comment]
We will try and get someone to test during beam off period

**Review by wesketchum — APPROVED** [review]
it's working for me at EHN1 as well in the dev area, so I'm approving.

**Pull request merged**